File: api/models/sell_request.py
from app import db
from event import Event
from user import User


class SellRequest(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    cre_dt = db.Column(db.DateTime)
    del_dt = db.Column(db.DateTime)
    event_id = db.Column(db.Integer, db.ForeignKey(Event.id))
    event = db.relationship(Event, uselist=False)
    submitted_by_user_id = db.Column(db.Integer, db.ForeignKey(User.id))
    submitted_by_user = db.relationship(User, uselist=False)
    price_min = db.Column(db.Integer)
    loc_long = db.Column(db.Float)
    loc_lat = db.Column(db.Float)
    loc_range = db.Column(db.Float)
    state = db.Column(db.String(80))

    def __init__(self):
        self.event = None
        self.price_min = None
        self.loc_long = None
        self.loc_lat = None
        self.loc_range = None
        self.state = None

File: mobile/app/models/sell_request.ts
import {Event} from './event'
import {UserPublic} from './user_public'

export class SellRequest {
    
    public id:number;
    public event:Event;
    public submitted_by_user:UserPublic;
    public price_min:number;
    public loc_long:number;
    public loc_lat:number;
    public loc_range:number;
    public state:string;
        
    constructor() {}
    
        public static fromJSON(jsonObj): SellRequest {
        var sell_request = new SellRequest();
        sell_request.id = jsonObj.id;
        sell_request.event = Event.fromJSON(jsonObj.event);
        sell_request.submitted_by_user = UserPublic.fromJSON(jsonObj.submitted_by_user);
        sell_request.price_min = jsonObj.price_min;
        sell_request.loc_long = jsonObj.loc_long;
        sell_request.loc_lat = jsonObj.loc_lat;
        sell_request.state = jsonObj.state;
        return sell_request
    }
}